gpx.cc

```diff
@@ -20,20 +20,28 @@
  */
 
 #include "defs.h"
-#include "cet_util.h"
 #include "garmin_fs.h"
 #include "garmin_tables.h"
+#include "src/core/datetime.h"
 #include "src/core/file.h"
 #include "src/core/logging.h"
 #include "src/core/xmlstreamwriter.h"
 #include "src/core/xmltag.h"
 
-#include <QtCore/QDateTime>
-#include <QtCore/QDebug>
-#include <QtCore/QRegExp>
-#include <QtCore/QXmlStreamReader>
+#include <QtCore/QDateTime>            // for QDateTime, QDate, QTime
+#include <QtCore/QHash>                // for QHash
+#include <QtCore/QList>                // for QList
+#include <QtCore/QString>              // for QString, QStringLiteral, QStringRef, QStaticStringData, QLatin1String, operator+
+#include <QtCore/QStringList>          // for QStringList
+#include <QtCore/QXmlStreamAttributes> // for QXmlStreamAttributes
+#include <QtCore/QXmlStreamNamespaceDeclarations> // for QXmlStreamNamespaceDeclarations
+#include <QtCore/QXmlStreamReader>     // for QXmlStreamReader, QXmlStreamReader::TokenType::Characters, QXmlStreamReader::TokenType::EndDocument, QXmlStreamReader::TokenType::EndElement, QXmlStreamReader::TokenType::Invalid, QXmlStreamReader::TokenType::StartElement
```

Comment:
You must have some tool that's better at this than what I'm using. This is great if it's auto-generated.

Reply:
include-what-you-use is the tool, with mapping files for Qt 5.9 that I generated.  But it is imperfect, so I have been using it in a piecemeal fashion and massaging the results by hand.  If you want to generate Qt mapping files I have a script or two to help automate that step.